src/NHSD.BuyingCatalogue.Organisations.Api/Data/ApplicationDbContext.cs

```diff
@@ -18,6 +18,16 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             if (modelBuilder is null)
                 throw new ArgumentNullException(nameof(modelBuilder));
 
+            modelBuilder.Entity<RelatedOrganisation>()
+                         .HasOne(or => or.ChildOrganisation)
+                         .WithMany()
+                         .HasForeignKey(or => or.RelatedOrganisationId);
+
+            modelBuilder.Entity<RelatedOrganisation>()
+                        .HasOne(or => or.Organisation)
+                        .WithMany(o => o.RelatedOrganisations)
+                        .HasForeignKey(or => or.OrganisationId);
+
             base.OnModelCreating(modelBuilder);
```

Comment:
i was trying to keep away from setting up a DbSet for RelatedOrgs

in reality, i would have loved this table and it's object to never really be seen at all and just get a pure list of orgs, but that's not how EF likes to do things.  

this code seems to be the EF way of doing code first linked tables. EF core can automagically work it out from Table Names, but then you have to name your silly stuff, so instead of RelatedOrganisation table, we would have OrganisationOrganisation table which would be super confusing. this is kind of the best way for EF to automatically pick it up without doing crazy stuff.

then again, i could hand roll it, but you get the advantage this way of being able to just add and remove the RelatedOrganisation from the list inside of an Organisation and it handles the adding/deletion pretty simply.

tbh, i don't mind either way, this just seemed to be the more EF way of things.

Reply:
We hand-roll the entity set-up in the ordering API so feel free to consider that option.